### PR TITLE
sys-cluster/teleport: add teleport v.2.2.4

### PR DESCRIPTION
**Pull request by gclawes**
Package-Manager: Portage-2.3.6, Repoman-2.3.2

### CODE REVIEW

**Comment by gclawes** [comment]
Symbol stripping is set by the upstream project's Makefile: https://github.com/gravitational/teleport/blob/master/Makefile#L15

From `go link` [documentation](https://golang.org/cmd/link/):
```
-s
	Omit the symbol table and debug information.

-w
	Omit the DWARF symbol table.
```

I'll speak to the upstream devs about changing this, but if it requires changes to the Makefile it will likely only be fixed in future releases.  Would a valid alternative be to patch the Makefile in `src_prepare()`?

**Comment by mgorny** [comment]
Yes, patching is fine. Upstream is more likely to accept a patch if you make it conditional somehow. It would be even helpful if you e.g. split `STRIP_OPTION=-s` so you could pass `emake STRIP_OPTION=` to wipe it.

**Comment by gclawes** [comment]
That, or moving `ADDFLAGS` to the end of the statement so it can override `-ldflags` may work.  I'll test it with a patch and suggest it to upstream devs.

**Comment by mgorny** [comment]
Ping. Do you need some help?

**Comment by gclawes** [comment]
Sorry, this dropped to the bottom of my to-do list.

I've fixed this in c7a00a4fab67003793d3ac002159ec41f037c4e7 by patching the Makefile in older versions.

Upstream dev has issued a fix for this, which will be included in future releases: https://github.com/gravitational/teleport/commit/e51eac1daa4ef2875ba5866fc5b71c00ca635736

**Comment by gclawes** [comment]
It looks like since this PR has been opened there has been an additional bugfix release (2.2.4).  I'll rename this PR for 2.2.4 and change the version in the MR, but the branch name will be the same.

**Comment by gclawes** [comment]
rebased on advice of QA bot.

**Comment by gclawes** [comment]
Removed 2.2.3

File: sys-cluster/teleport/files/Makefile.patch [hunk 1]
```diff
@@ -0,0 +1,11 @@
+--- src/github.com/gravitational/teleport/Makefile
```
Comment (mgorny) [inline sys-cluster/teleport/files/Makefile.patch]: This is a bad patch name. We usually prefix patches with `${P}-`, especially if they're likely to apply to one specific version only. It makes the updates and cleanup easier.

File: sys-cluster/teleport/teleport-2.2.0.ebuild [hunk 2]
```diff
@@ -27,8 +27,15 @@ DEPEND="
 	>=dev-lang/go-1.8.3"
 RDEPEND=""
 
+src_prepare() {
+	default
+
+	cp "${FILESDIR}"/Makefile.patch "${S}"
```
Comment (mgorny) [inline sys-cluster/teleport/teleport-2.2.0.ebuild]: This is not necessary. You can apply patches straight from FILESDIR.

File: sys-cluster/teleport/teleport-2.2.0.ebuild [hunk 3]
```diff
@@ -27,8 +27,15 @@ DEPEND="
...
+	epatch "${S}"/Makefile.patch
```
Comment (mgorny) [inline sys-cluster/teleport/teleport-2.2.0.ebuild]: eapply.

File: sys-cluster/teleport/teleport-2.2.4.ebuild [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+EAPI=6
+inherit eutils golang-build systemd user
+
+DESCRIPTION="Modern SSH server for teams managing distributed infrastructure"
+HOMEPAGE="https://gravitational.com/teleport"
+
+EGO_PN="github.com/gravitational/${PN}/..."
+
+if [ ${PV} == "9999" ] ; then
```
Comment (mgorny) [inline sys-cluster/teleport/teleport-2.2.4.ebuild]: Use `[[ ... ]]` in bash, please.

File: sys-cluster/teleport/teleport-2.2.0.ebuild [hunk 5]
```diff
@@ -27,8 +27,15 @@ DEPEND="
 	>=dev-lang/go-1.8.3"
```
Comment (mgorny) [inline sys-cluster/teleport/teleport-2.2.0.ebuild]: Besides, it's your choice but normally you don't have to patch old versions, just the newest.

File: sys-cluster/teleport/files/Makefile.patch (hunk 1 above)
Comment (gclawes) [inline sys-cluster/teleport/files/Makefile.patch]: fixed, renamed to `teleport-makefile-buildflags.patch`

File: sys-cluster/teleport/teleport-2.2.0.ebuild (hunk 5 above)
Comment (gclawes) [inline sys-cluster/teleport/teleport-2.2.0.ebuild]: renamed patch, applied to all versions

File: sys-cluster/teleport/teleport-2.2.0.ebuild (hunk 2 above)
Comment (gclawes) [inline sys-cluster/teleport/teleport-2.2.0.ebuild]: removed `src_prepare()`, used `PATCHES`

File: sys-cluster/teleport/teleport-2.2.0.ebuild (hunk 3 above)
Comment (gclawes) [inline sys-cluster/teleport/teleport-2.2.0.ebuild]: removed `src_prepare()`, used `PATCHES`

File: sys-cluster/teleport/teleport-2.2.4.ebuild (hunk 4 above)
Comment (gclawes) [inline sys-cluster/teleport/teleport-2.2.4.ebuild]: fixed

**Pull request closed without merging**